### DISCUSSION TITLE
Question about Native Image debugging

### DISCUSSION THREAD

**Discussion by SetoKaiba** [quarkusio/quarkus#37514] [Q&A] [answered]
The doc only mentioned that the `quarkus.native.debug.enabled`. What does it do actually? Does it add the `-g` to `javac`? Does it add the `-g` to `native-image`? Does it add the `-O0,-H:-OmitInlinedMethodDebugLineInfo` to `native-image`?

[a comment by an automated account is not shown]

**Reply by zakkak** [reply, marked as answer]
Hello @SetoKaiba, 

> What does it do actually?

That depends on the Quarkus and Mandrel/GraalVM version.

To see what actually happens follow the `io.quarkus.deployment.pkg.NativeConfig.Debug#enabled` references in the source code, e.g.:

https://github.com/quarkusio/quarkus/blob/45a7c9dee6fb49f3c4e4f117af05a350d94ede2c/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java#L746-L758

> Does it add the -g to javac? 

No, but there is a feature request for this, see https://github.com/quarkusio/quarkus/issues/31218

> Does it add the -g to native-image? 

Yes

> Does it add the -O0,-H:-OmitInlinedMethodDebugLineInfo to native-image?

No, we don't want to pass `-O0` since we want users to debug an instance of their application as close to what they will deploy as possible. In our opinion `-O0` should only be used when necessary.

`-H:-OmitInlinedMethodDebugLineInfo` is also not passed by Quarkus, note that starting with Mandrel 23.1 / GraalVM for JDK 21 it's no longer needed, it's the default.
